code/modules/vehicles/aircraft.dm

```diff
@@ -0,0 +1,1297 @@
+	for(var/datum/atom_hud/squad/aircraft_hud in GLOB.huds) //Add to the squad HUD
+		aircraft_hud.add_to_hud(src)
+	current_fuel = max_fuel
+	if(starting_weapons)
+		attach_starting_weapon()
+	if(deployable && !deployed)
+		deploy(null, TRUE)
+	integrity_failure = max_integrity/5	//20% of HP triggers damage alert
+	light_smoke_effect = new(src, /particles/aircraft_smoke_light)
+	heavy_smoke_effect = new(src, /particles/aircraft_smoke_heavy)
+	light_smoke_effect.pixel_x = 16
+	heavy_smoke_effect.pixel_x = 16
+	update_icon()
+
+/obj/vehicle/sealed/aircraft/update_icon()
+	show_aircraft_health()
+	show_aircraft_fuel()
+	handle_offsets()
+	if(obj_integrity > integrity_failure && obj_integrity < max_integrity * 0.8)	//From 20% to 70% HP, some smoke will emit
+		light_smoke_effect.particles.spawning = initial(light_smoke_effect.particles.spawning)
+		heavy_smoke_effect.particles.spawning = 0
+	else if(obj_integrity <= integrity_failure)	//20% HP and under will produce thick, black smoke
+		heavy_smoke_effect.particles.spawning = initial(heavy_smoke_effect.particles.spawning)
+		light_smoke_effect.particles.spawning = 0
+	else
+		light_smoke_effect.particles.spawning = 0
+		heavy_smoke_effect.particles.spawning = 0
+	light_smoke_effect.pixel_z = pixel_z
+	heavy_smoke_effect.pixel_z = pixel_z
+	light_smoke_effect.layer = layer + 0.01
+	heavy_smoke_effect.layer = layer + 0.01
+	return ..()
+
+///Proc for delayed turning
+/obj/vehicle/sealed/aircraft/proc/turn_while_flying(direction)
+	if(direction == dir)
+		return
+	if(!COOLDOWN_CHECK(src, turning_cooldown))
+		return
+	COOLDOWN_START(src, turning_cooldown, turn_delay)
+	if(direction == REVERSE_DIR(dir))
+		return setDir(turn(dir, pick(45, -45)))
+	var/angle1 = dir2angle(direction)
+	var/angle2 = dir2angle(dir)
+	var/angle_to_turn = closer_angle_difference(angle1, angle2) > 0 ? 45 : -45
+	setDir(turn(dir, angle_to_turn))
+	if(direction != dir)
+		addtimer(CALLBACK(src, .proc/turn_while_flying, direction), turn_delay)
+
+/obj/vehicle/sealed/aircraft/setDir(newdir)
+	. = ..()
+	handle_offsets()
+
+///Adjust the sprite positions when turning
+/obj/vehicle/sealed/aircraft/proc/handle_offsets()
+	switch(dir)
+		if(NORTH)
+			pixel_x = offsets[1]
+			pixel_y = offsets[2]
+		if(NORTHEAST)
+			pixel_x = offsets[3]
+			pixel_y = offsets[4]
+		if(EAST)
+			pixel_x = offsets[5]
+			pixel_y = offsets[6]
+		if(SOUTHEAST)
+			pixel_x = offsets[7]
+			pixel_y = offsets[8]
+		if(SOUTH)
+			pixel_x = offsets[9]
+			pixel_y = offsets[10]
+		if(SOUTHWEST)
+			pixel_x = offsets[11]
+			pixel_y = offsets[12]
+		if(WEST)
+			pixel_x = offsets[13]
+			pixel_y = offsets[14]
+		if(NORTHWEST)
+			pixel_x = offsets[15]
+			pixel_y = offsets[16]
+	if(LAZYLEN(weapon_overlays))
+		//Remove the overlays and re-apply them in the direction the aircraft is facing
+		cut_overlays()
+		weapon_overlays.Cut()
+		for(var/obj/item/weapon/gun/aircraft/gun AS in attached_weapons)
+			var/weapon_position_in_list = attached_weapons.Find(gun)
```

Comment:
why do you switch spam

Reply:
Is there a better way? It's to check for the direction, like all other procs do it.